src/components/select/bl-select.ts

```diff
@@ -0,0 +1,402 @@
+   */
+  @property({})
+  placeholder?: string;
+
+  /**
+   * Sets the size value. Select component's height value will be changed accordingly
+   */
+  @property({ type: String, reflect: true })
+  size: SelectSize = 'medium';
+
+  /**
+   * When option is not selected, shows component in error state
+   */
+  @property({ type: Boolean })
+  required = false;
+
+  /**
+   * Shows the component in disabled state.
+   */
+  @property({ type: Boolean })
+  disabled = false;
+
+  /**
+   * Allows multiple options to be selected
+   */
+  @property({ type: Boolean })
+  multiple = false;
+
+  /**
+   * Makes label as fixed positioned
+   */
+  @property({ type: Boolean, attribute: 'label-fixed' })
+  labelFixed = false;
+
+  /**
+   * Adds help text
+   */
+  @property({ type: String, attribute: 'help-text' })
+  helpText?: string;
+
+  /**
+   * Set custom error message
+   */
+  @property({ type: String, attribute: 'invalid-text' })
+  customInvalidText?: string;
+
+  /* Declare internal reactive properties */
+  @state()
+  private _isMenuOpen = false;
+
+  @state()
+  private _selectedOptions: ISelectOption[] = [];
+
+  @state()
+  private _additionalSelectedOptionCount = 0;
+
+  @state()
+  private _isInvalid = false;
+
+  @query('.selected-options')
+  private _selectedOptionsContainer!: HTMLElement;
+
+  @queryAll('.selected-options li')
+  private _selectedOptionsItems!: Array<HTMLElement>;
+
+  @query('.select-menu')
+  private _selectMenu: HTMLElement;
+
+  @query('.select-input')
+  private _selectInput: HTMLElement;
+
+  @event('bl-select') private _onBlSelect: EventDispatcher<ISelectOption[]>;
+
+  private _connectedOptions: BlSelectOption[] = [];
+
+  get options() {
+    return this._connectedOptions;
+  }
+
+  get isMenuOpen() {
+    return this._isMenuOpen;
+  }
+
+  get selectedOptions() {
+    return this._selectedOptions;
+  }
+
+  get additionalSelectedOptionCount() {
+    return this._additionalSelectedOptionCount;
+  }
+
+  get isInvalid() {
+    return this._isInvalid;
+  }
+
+  private _clickOutsideHandler = (event: MouseEvent) => {
+    const target = event.target as HTMLElement;
+
+    if (!this.contains(target) && this._isMenuOpen) {
+      this._isMenuOpen = false;
+      this._checkRequired();
+    }
+  };
+
+  private _showSelectMenu() {
+    const defaultMaxHeight = parseInt(
+      getComputedStyle(this._selectMenu)
+        .getPropertyValue('--bl-select-menu-height')
+        .replace('px', '')
+    );
+
+    computePosition(this._selectInput, this._selectMenu, {
+      placement: 'bottom',
+      strategy: 'absolute',
+      middleware: [
+        flip(),
+        offset(8),
+        size({
+          apply(args: MiddlewareArguments & { availableHeight: number }) {
+            Object.assign(args.elements.floating.style, {
+              maxHeight: `${
+                args.availableHeight > defaultMaxHeight ? defaultMaxHeight : args.availableHeight
+              }px`,
```

Comment:
Why we are not just setting a `max-height` in CSS?

Reply:
Let's think a container there is not any available space below and has limited space above. If we don't limit the max height to available height, some options will not visible as seen in the example below. 

set `max-height` to available height
<img width="350" alt="image" src="https://user-images.githubusercontent.com/49120729/190181340-3487f696-4bd4-4142-a91b-d28b11461c4f.png">
only `max-height` in css
<img width="350" alt="image" src="https://user-images.githubusercontent.com/49120729/190182453-2b01301e-fd01-4de3-ad7e-d1480d29db46.png">

Reply:
If I understand correctly, you mean this is related using absolute position relative to parent. Maybe we need to consider using fixed positioning (as I mentioned in another thread). In that case you may re-consider here to simplify logic.

Reply:
Note: We found a solution for fixed positioning problems. We changed positioning to `fixed` with the help of autoUpdate function in floating-ui package.